aws/data_source_aws_iam_policy_document_test.go

```diff
@@ -155,25 +156,42 @@ func TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_Strin
 			{
 				Config: testAccAWSIAMPolicyDocumentConfigStatementPrincipalIdentifiersStringAndSlice,
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(dataSourceName, "json", testAccAWSIAMPolicyDocumentExpectedJSONStatementPrincipalIdentifiersStringAndSlice),
+					resource.TestCheckResourceAttr(dataSourceName, "json", testAccAWSIAMPolicyDocumentExpectedJSONStatementPrincipalIdentifiersStringAndSlice()),
 				),
 			},
 		},
 	})
 }
 
 // Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/10777
-func TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_MultiplePrincipals(t *testing.T) {
+func TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_MultiplePrincipalsAWS(t *testing.T) {
 	dataSourceName := "data.aws_iam_policy_document.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionPreCheck(t, "aws") },
```

Comment:
Use [`endpoints.AwsPartitionID`](https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#pkg-constants)?

Reply:
Excellent! We should be using the constants for service IDs too (eg, `testAccPartitionHasServicePreCheck(endpoints.FirehoseServiceID, t)`).

Reply:
After change (GovCloud):

```
   provider_test.go:656: skipping tests; current partition (aws-us-gov) does not equal aws
--- SKIP: TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_MultiplePrincipalsAWS (1.53s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_MultiplePrincipalsGov (18.88s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_noStatementMerge (18.88s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_StringAndSlice (18.88s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_override (18.88s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_sourceConflicting (18.89s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_basic (18.90s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_noStatementOverride (18.93s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_duplicateSid (19.85s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_Version_20081017 (27.48s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_source (30.22s)
```

Commercial:
```
    provider_test.go:656: skipping tests; current partition (aws) does not equal aws-us-gov
--- SKIP: TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_MultiplePrincipalsGov (0.89s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_MultiplePrincipalsAWS (13.91s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_override (13.98s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_sourceConflicting (14.09s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_basic (14.09s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_noStatementMerge (14.13s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_noStatementOverride (14.13s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_Statement_Principal_Identifiers_StringAndSlice (14.15s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_duplicateSid (15.03s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_Version_20081017 (19.64s)
--- PASS: TestAccAWSDataSourceIAMPolicyDocument_source (21.66s)
```

Reply:
I'd suggest being careful with the service constants, they are deprecated in the `endpoints` package.

> Deprecated: Use client package's EndpointsID value instead of these ServiceIDs. These IDs are not maintained, and are out of date.